managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -249,6 +250,10 @@ public ManagedLedgerImpl(ManagedLedgerFactoryImpl factory, BookKeeper bookKeeper
         this.config = config;
         this.store = store;
         this.name = name;
+        this.ledgerMetadata = ImmutableMap.of(
+                "application", "pulsar".getBytes(StandardCharsets.UTF_8),
+                "component", "managed-ledger".getBytes(StandardCharsets.UTF_8),
+                "topic", name.getBytes(StandardCharsets.UTF_8));
```

Comment:
At this point the name is not really the "topic name" but rather an encoding of it. Eg: some parts of the topic names are swapped in position (there's a reason for that :) ), and also the string is url-encoded to avoid issues with ZK paths (eg: `/` need to be escaped). 

For that, I'd rather calling that `managed-ledger` instead of topic to avoid confusion. We already have tools to convert <-> the two different forms anyway.